### PR TITLE
docs: recommend enabling `fetch` support in HttpClient when using SSR

### PR DESCRIPTION
**Pull request by AndrewKushnir**
Related to https://github.com/angular/angular/pull/52037.

## PR Type
What kind of change does this PR introduce?

- [x] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/common/http/src/provider.ts [hunk 1]
```diff
@@ -57,6 +57,18 @@ function makeHttpFeature<KindT extends HttpFeatureKind>(
  * feature functions to `provideHttpClient`. For example, HTTP interceptors can be added using the
  * `withInterceptors(...)` feature.
  *
+ * <div class="alert is-helpful">
+ *
+ * It's strongly recommended to enable `fetch` for applications that use Server-Side Rendering
```
Comment (alan-agius4) [inline packages/common/http/src/provider.ts:62]: Nit maybe link fetch to https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API

**Review by alan-agius4 — APPROVED** [review]
LGTM, optional NIT.

File: packages/common/http/src/provider.ts (hunk 1 above)
Comment (AndrewKushnir) [reply packages/common/http/src/provider.ts]: Good idea, added the link 👍

**Comment by AndrewKushnir** [comment]
**Caretaker note**: docs-only change, no presubmit is required.

**Comment by atscott** [comment]
This PR was merged into the repository by commit 22fa9fe488dd72b2f65b8a3951d492c1f178b387.

**Pull request closed without merging**